xknx/management/procedures.py

```diff
@@ -58,3 +66,115 @@ async def nm_individual_address_check(
         # if Disconnect is received immediately, IA is occupied
         logger.debug("Device does not support transport layer connections. %s", ex)
         return True
+
+
+async def nm_individual_address_read(
+    xknx: XKNX, timeout: float | None = 3, raise_if_multiple: bool = False
+) -> list[IndividualAddress]:
+    """
+    Request individual addresses of all devices that are in programming mode. If multiple devices are in programming mode, a ManagementConnectionError is raised.
+
+    :param: timeout specifies the timeout in seconds, the KNX specification requires a timeout of 3s.
+    """
+
+    addresses = []
+    # initialize queue or event handler gathering broadcasts
+    async with xknx.management.broadcast() as bc_context:
+        broadcast_telegram = Telegram(
+            GroupAddress("0/0/0"), payload=apci.IndividualAddressRead()
+        )
+        await xknx.management.send_broadcast(broadcast_telegram)
+        async for result in bc_context.receive(timeout=timeout):
+            if isinstance(result.payload, apci.IndividualAddressResponse):
+                addresses.append(result.source_address)
+                if raise_if_multiple and (len(addresses) > 1):
+                    raise ManagementConnectionError(
+                        "More than one KNX device is in programming mode."
+                    )
+    return addresses
+
+
+async def nm_invididual_address_write(
+    xknx: XKNX, individual_address: IndividualAddressableType
+) -> bool:
+    """
+    Write the individual address of a single device in programming mode.
+
+    TODO: detail exceptions if this failed.
+    """
```

Comment:
```suggestion
    """Write the individual address of a single device in programming mode."""
```
I think we got it covered 😉